Migration of Kafka source configuration.

### Apache StreamPipes version

dev (current development state)

### Affected StreamPipes components

Backend, Connect, Processing Elements, UI

### What happened?


StreamPipes 0.93.0-SNAPSHOT has new attributes to configure a Kafka source like "Auto Offest Reset". If the configuration was created in 0.92.0 this attributes could not be accessed[1],  which leads to errors in the UI and during runtime. 

What do you think of the idea to use a default value if no other value is provided? If I see it correctly changing the value should still be possible.



[1]
`
streampipes-extensions-all-jvm-1           | Caused by: java.lang.NullPointerException: Cannot invoke "org.apache.streampipes.model.staticproperty.StaticPropertyAlternatives.getAlternatives()" because "alternatives" is null
streampipes-extensions-all-jvm-1           | 	at org.apache.streampipes.sdk.extractor.AbstractParameterExtractor.selectedAlternativeInternalId(AbstractParameterExtractor.java:588)
streampipes-extensions-all-jvm-1           | 	at org.apache.streampipes.pe.shared.config.kafka.kafka.KafkaConnectUtils.getConfig(KafkaConnectUtils.java:128)
streampipes-extensions-all-jvm-1           | 	at org.apache.streampipes.connect.iiot.protocol.stream.KafkaProtocol.resolveConfiguration(KafkaProtocol.java:117)
streampipes-extensions-all-jvm-1           | 	at org.apache.streampipes.extensions.management.api.RuntimeResolvableRequestHandler.handleRuntimeResponse(RuntimeResolvableRequestHandler.java:50)
streampipes-extensions-all-jvm-1           | 	at org.apache.streampipes.rest.extensions.connect.RuntimeResolvableResource.fetchConfigurations(RuntimeResolvableResource.java:60)
streampipes-extensions-all-jvm-1           | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
streampipes-extensions-all-jvm-1           | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
streampipes-extensions-all-jvm-1           | 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
streampipes-extensions-all-jvm-1           | 	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
streampipes-extensions-all-jvm-1           | 	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory.lambda$static$0(ResourceMethodInvocationHandlerFactory.java:52)`

### How to reproduce?

Configure a SP  0.93.0 instance with a Kafka client and try to start this adapter after upgrading to  0.93.0-SNAPSHOT.
Try to configure a new adapter when the UI configuration was not reloaded from the extension service. (Should the reload part of the upgrade process?)

### Expected behavior

_No response_

### Additional technical information

_No response_

### Are you willing to submit a PR?

Yes, I am willing to submit a PR!

--- a/streampipes-extensions/streampipes-pipeline-elements-shared/src/main/java/org/apache/streampipes/pe/shared/config/kafka/kafka/KafkaConnectUtils.java
+++ b/streampipes-extensions/streampipes-pipeline-elements-shared/src/main/java/org/apache/streampipes/pe/shared/config/kafka/kafka/KafkaConnectUtils.java
@@ -26,6 +26,7 @@ import org.apache.streampipes.messaging.kafka.security.KafkaSecuritySaslSSLConfi
 import org.apache.streampipes.messaging.kafka.security.KafkaSecurityUnauthenticatedPlainConfig;
 import org.apache.streampipes.messaging.kafka.security.KafkaSecurityUnauthenticatedSSLConfig;
 import org.apache.streampipes.model.staticproperty.StaticPropertyAlternative;
+import org.apache.streampipes.model.staticproperty.StaticPropertyAlternatives;
 import org.apache.streampipes.sdk.StaticProperties;
 import org.apache.streampipes.sdk.helpers.Alternatives;
 import org.apache.streampipes.sdk.helpers.Label;
@@ -123,12 +124,20 @@ public class KafkaConnectUtils {
           new KafkaSecurityUnauthenticatedPlainConfig();
     }
 
+    StaticPropertyAlternatives alternatives = extractor.getStaticPropertyByName(AUTO_OFFSET_RESET_CONFIG,
+            StaticPropertyAlternatives.class);
 
+    // Set default value if no value is provided.
+    if (alternatives == null) {
+      AutoOffsetResetConfig autoOffsetResetConfig = new AutoOffsetResetConfig(KafkaConnectUtils.LATEST);
 
-    String auto = extractor.selectedAlternativeInternalId(AUTO_OFFSET_RESET_CONFIG);
-    AutoOffsetResetConfig autoOffsetResetConfig = new AutoOffsetResetConfig(auto);
+      return new KafkaConfig(brokerUrl, port, topic, securityConfig, autoOffsetResetConfig);
+    } else {
+      String auto = extractor.selectedAlternativeInternalId(AUTO_OFFSET_RESET_CONFIG);
+      AutoOffsetResetConfig autoOffsetResetConfig = new AutoOffsetResetConfig(auto);
 
-    return new KafkaConfig(brokerUrl, port, topic, securityConfig, autoOffsetResetConfig);
+      return new KafkaConfig(brokerUrl, port, topic, securityConfig, autoOffsetResetConfig);
+    }
   }
 
   private static boolean isUseSSL(String authentication) {

